### PR TITLE
Song Queue feature

### PR DESCRIPTION
**Pull request by sonnynomnom**
PR to build the Playlist aka Song Queue feature. Here's the design from Guyen:

playlist idea

- [x] Build the Song Queue div in **room.html**, **queue.js**, and **queue.css**.
- [x] Add a slide in animation.
- [x] Update **songs.js** with a `duration` param.
- [ ] Push `karaoke-video` a little to the left when the Song Queue div slide in.
- [ ] Set height of Song Queue div the same as `karaoke-video`.
- [ ] Add styling for current song:
  - [ ] Yellow text
  - [ ] Progress bar circle
  - [ ] Album spin / rotate
- [ ] Add fade in the Song Queue div.

### CODE REVIEW

File: socketController.js [hunk 1]
```diff
@@ -246,12 +248,24 @@ module.exports = (io) => {
...
+     socket.on(SE_ADD_SONG_TO_QUEUE, function (queueSong) {
+      console.log(
+        `Client ${socket.id} in room ${roomName} emitted 'addSongToQueue' event with ${queueSong["title"]}`
+      );
+
+      rooms[roomName].playInfo.queue.push(queueSong);
+
+      socket.to(roomName).emit(SE_ADD_SONG_TO_QUEUE, queueSong);
+    });
+
```
Comment (codecaaron) [inline socketController.js:269]: Does the queue remove songs after they've stopped from display?

File: public/styles/queue.css [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+.queue-album {
+  height: 35px;
+  width: 35px;
+}
+
+.queue-album img {
+  border-radius: 50%;
+  animation: rotation 5s infinite linear;
+}
```
Comment (codecaaron) [inline public/styles/queue.css:73]: Add a class for the image.  1 Level of specificity per selector (except in cases where structure requires it like here https://github.com/karaokenite/karaokenite-glitch/pull/17/files#diff-d4ccf3b9b387dc0fc72f0695246155525703377e1bddc7ce13bee7c0177fb470R61)

File: public/styles/queue.css [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+.queue-album-circle {
+  position: relative;
+  z-index: 99;
+  left: 13px; 
+  top: -28px;
+  border-radius: 100%;
+  width: 8px; 
+  height: 8px;
```
Comment (codecaaron) [inline public/styles/queue.css:91]: This is pretty brittle - let's do this with relative values or flex positioning so if we ever change the size of one thing this just works :)

File: public/styles/queue.css [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  background-color: rgba(0, 0, 0, 0.1);
+  backdrop-filter: blur(80px);
+}
+
+.queue-title {
+  display: grid;
+  font-weight: 500;
+  color: #F2F2F2;
```
Comment (codecaaron) [inline public/styles/queue.css:99]: Do we have color vars for this?

File: public/styles/queue.css [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  min-height: 278px;
+}
+
+.queue-list {
+  display: flex;
+  padding: 5px;
+  position: relative;
+  flex-direction: column;
+  max-height: 300px;
+  overflow-y: scroll;
```
Comment (codecaaron) [inline public/styles/queue.css:32]: Auto works here and will add a scrollbar only when necessary.
```suggestion
  overflow-y: auto;
```

**Review by codecaaron — COMMENTED** [review]
Nice - looking good, easy to read through!

File: public/js/queue.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        <div class="queue-group">
+          <div>
+            <div class="queue-title">${song.title}</div>
+            <div class="queue-artist">${song.artist}</div>
+          </div>
+          <div class="queue-time">${song.length}</div>
+        </div>
+      </div>
+    `;
+    })
```
[the remark this answers is not in the document]
Comment (sonnynomnom) [reply public/js/queue.js:24]: Oh niiice!

File: public/room.html [hunk 7]
```diff
@@ -102,13 +108,52 @@ <h1 class="brand-logo">
...
+                  />
+                  <div class="queue-group">
+                    <div>
+                      <div class="queue-title">Welcome Video</div>
+                      <div class="queue-artist">Karaoke Nite</div>
+                    </div>
+                    <div class="queue-time">0:10</div>
+                  </div>
+                </div>
+
```
[the remark this answers is not in the document]
Comment (sonnynomnom) [reply public/room.html:138]: Deleted!

File: public/styles/queue.css (hunk 4 above)
Comment (sonnynomnom) [reply public/styles/queue.css:99]: Nope. Just updated the colors!

Screen Shot 2022-01-17 at 3 23 42 PM

**Review by codecaaron — APPROVED** [review]

**Pull request merged**